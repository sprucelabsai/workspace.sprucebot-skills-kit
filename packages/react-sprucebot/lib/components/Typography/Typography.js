'use strict';

Object.defineProperty(exports, "__esModule", {
	value: true
});
exports.SectionHeading = exports.A = exports.H6 = exports.H5 = exports.H4 = exports.H3 = exports.H2 = exports.H1 = exports.Paragraph = undefined;

var _styledComponents = require('styled-components');

var _styledComponents2 = _interopRequireDefault(_styledComponents);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var Paragraph = exports.Paragraph = (0, _styledComponents2.default)('p').attrs({
	className: function className(props) {
		return props.fine ? 'fine__print' : '';
	}
}).withConfig({
	displayName: 'Typography__Paragraph',
	componentId: 'sc-1caqefb-0'
})(['']);

var H1 = exports.H1 = (0, _styledComponents2.default)('h1').attrs({
	className: function className(props) {
		return props.with_subheader ? 'with__subheader' : '';
	}
}).withConfig({
	displayName: 'Typography__H1',
	componentId: 'sc-1caqefb-1'
})(['']);
var H2 = exports.H2 = (0, _styledComponents2.default)('h2').attrs({
	className: function className(props) {
		return props.subheader ? 'is__subheader' : '';
	}
}).withConfig({
	displayName: 'Typography__H2',
	componentId: 'sc-1caqefb-2'
})(['']);
var H3 = exports.H3 = (0, _styledComponents2.default)('h3').withConfig({
	displayName: 'Typography__H3',
	componentId: 'sc-1caqefb-3'
})(['']);
var H4 = exports.H4 = (0, _styledComponents2.default)('h4').withConfig({
	displayName: 'Typography__H4',
	componentId: 'sc-1caqefb-4'
})(['']);
var H5 = exports.H5 = (0, _styledComponents2.default)('h5').withConfig({
	displayName: 'Typography__H5',
	componentId: 'sc-1caqefb-5'
})(['']);
var H6 = exports.H6 = (0, _styledComponents2.default)('h6').withConfig({
	displayName: 'Typography__H6',
	componentId: 'sc-1caqefb-6'
})(['']);

var A = exports.A = (0, _styledComponents2.default)('a').withConfig({
	displayName: 'Typography__A',
	componentId: 'sc-1caqefb-7'
})(['']);
var SectionHeading = exports.SectionHeading = (0, _styledComponents2.default)('h2').attrs({
	className: 'profile__subtitle'
}).withConfig({
	displayName: 'Typography__SectionHeading',
	componentId: 'sc-1caqefb-8'
})(['']);