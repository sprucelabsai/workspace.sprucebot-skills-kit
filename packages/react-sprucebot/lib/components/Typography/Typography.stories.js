'use strict';

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _react3 = require('@storybook/react');

var _addonInfo = require('@storybook/addon-info');

var _react4 = require('@storybook/addon-knobs/react');

var _storybookReadme = require('storybook-readme');

var _Typography = require('./Typography');

var _Typography2 = require('./Typography.md');

var _Typography3 = _interopRequireDefault(_Typography2);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var stories = (0, _react3.storiesOf)('Typography', module);
stories.addDecorator(_react4.withKnobs);

stories.add('Interactive', (0, _storybookReadme.withReadme)(_Typography3.default, function () {
	return _react2.default.createElement(
		'div',
		null,
		_react2.default.createElement(
			_Typography.H1,
			null,
			'Heading One'
		),
		_react2.default.createElement(
			_Typography.H2,
			null,
			'Heading Two'
		),
		_react2.default.createElement(
			_Typography.H3,
			null,
			'Heading Three'
		),
		_react2.default.createElement(
			_Typography.H4,
			null,
			'Heading Four'
		),
		_react2.default.createElement(
			_Typography.H5,
			null,
			'Heading Five'
		),
		_react2.default.createElement(
			_Typography.H6,
			null,
			'Heading Six'
		),
		_react2.default.createElement(
			_Typography.SectionHeading,
			null,
			'Section Heading'
		),
		_react2.default.createElement(
			_Typography.Paragraph,
			null,
			'This is body Copy'
		),
		_react2.default.createElement(
			_Typography.A,
			null,
			'Link'
		)
	);
}));