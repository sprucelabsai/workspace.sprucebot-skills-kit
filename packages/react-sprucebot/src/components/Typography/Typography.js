import React from 'react'

export const Paragraph = ({ className, fine, ...props }) => {
	return (
		<p
			{...props}
			className={`${fine ? 'fine__print' : ''} ${className || ''}`}
		/>
	)
}

export const H1 = ({ className, with_subheader, ...props }) => {
	return (
		<h1
			{...props}
			className={`${with_subheader ? 'with__subheader' : ''} ${className ||
				''}`}
		/>
	)
}

export const H2 = ({ className, subheader, ...props }) => {
	return (
		<h2
			{...props}
			className={`${subheader ? 'is__subheader' : ''} ${className || ''}`}
		/>
	)
}

export const H3 = ({ className, ...props }) => {
	return <h3 {...props} className={`${className || ''}`} />
}

export const H4 = ({ className, ...props }) => {
	return <h4 {...props} className={`${className || ''}`} />
}

export const H5 = ({ className, ...props }) => {
	return <h5 {...props} className={`${className || ''}`} />
}

export const H6 = ({ className, ...props }) => {
	return <h6 {...props} className={`${className || ''}`} />
}

export const A = ({ className, ...props }) => {
	return <a {...props} className={`${className || ''}`} />
}

export const SectionHeading = ({ className, ...props }) => {
	return <h2 {...props} className={`profile__subtitle ${className || ''}`} />
}
