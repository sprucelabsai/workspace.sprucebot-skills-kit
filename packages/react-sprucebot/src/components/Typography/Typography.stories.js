import React from 'react'
import { storiesOf } from '@storybook/react'
import { withInfo } from '@storybook/addon-info'
import { withKnobs } from '@storybook/addon-knobs/react'
import { withReadme } from 'storybook-readme'
import {
	Paragraph,
	H1,
	H2,
	H3,
	H4,
	H5,
	H6,
	A,
	SectionHeading
} from './Typography'
import readme from './Typography.md'

const stories = storiesOf('Typography', module)
stories.addDecorator(withKnobs)

stories.add(
	'Interactive',
	withReadme(readme, () => (
		<div>
			<H1>Heading One</H1>
			<H2>Heading Two</H2>
			<H3>Heading Three</H3>
			<H4>Heading Four</H4>
			<H5>Heading Five</H5>
			<H6>Heading Six</H6>
			<SectionHeading>Section Heading</SectionHeading>
			<Paragraph>This is body Copy</Paragraph>
			<A>Link</A>
		</div>
	))
)
