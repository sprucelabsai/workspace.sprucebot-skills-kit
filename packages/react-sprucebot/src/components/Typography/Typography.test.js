import React from 'react'
import { Paragraph, H1, H2, H3, H4, H5, H6, A } from './Typography'
import { shallow } from 'enzyme'

describe('H1 tests', () => {
	it('Should match the snapshot', () => {
		const renderedComponent = shallow(<H1 />)

		expect(renderedComponent).toMatchSnapshot()
	})
})
