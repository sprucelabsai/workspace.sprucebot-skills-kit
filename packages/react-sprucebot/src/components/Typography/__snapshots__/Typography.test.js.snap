// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`h1 tag it renders 1`] = `
<h1
  className=""
/>
`;

exports[`h1 tag it renders header and subheader 1`] = `
<div>
  <h1
    className=""
  />
  <h2
    className="is__subheader "
  />
</div>
`;

exports[`p tag it fine prints 1`] = `
<p
  className="fine__print "
/>
`;

exports[`p tag it renders 1`] = `
<p
  className=""
/>
`;
